purchase_delivery_invoice/purchase_delivery_invoice.py

```diff
@@ -164,14 +205,14 @@ def _prepare_shipping_invoice_line(
                     cr, uid, invoice.company_id.currency_id.id,
                     invoice.currency_id.id, price,
                     context=dict(context or {}, date=invoice.date_invoice))
-            account_id = picking.carrier_id.product_id. \
-                property_account_income.id
+                account_id = picking.carrier_id.product_id.property_account_income.id
+
             if not account_id:
                 account_id = picking.carrier_id.product_id.categ_id. \
                     property_account_income_categ.id
-            taxes = picking.carrier_id.product_id.taxes_id
-            partner = picking.partner_id or False
-            fp = invoice.fiscal_position or partner.property_account_position
+                taxes = picking.carrier_id.product_id.taxes_id
```

Comment:
Wrong indentation. It appears those variables were moved inside the if block by mistake. Please be careful with all changes done to module.